### PR TITLE
Tags in branch overview and tag fixes

### LINKED ISSUE #1081: Implement support for (lightweight) tags
As discussed here #960 support for tags would be useful in some cases. I'd like to start with implementing basic tag support. E.g. display, create and delete tags.
For the sake of simplicity I'd start with [lightweight tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging).

### PR DESCRIPTION
**Pull request by christianguenter2**
#1081 
- A proper error message appeȧrs when no tags exists.
- Display the SHA on the tag create popup
- Display tags in branch overview

### CODE REVIEW

**Comment by larshp** [comment]
looks like something messed up the commits, the background mode and 1.51.0 should not be part of this PR?

File: src/zabapgit_git.prog.abap [hunk 1]
```diff
@@ -602,6 +602,8 @@ CLASS lcl_git_pack IMPLEMENTATION.
         rv_type = zif_abapgit_definitions=>gc_type-tree.
       WHEN '011'.
         rv_type = zif_abapgit_definitions=>gc_type-blob.
+      WHEN '100'.
+        rv_type = zif_abapgit_definitions=>gc_type-tag.
```
Comment (larshp) [inline src/zabapgit_git.prog.abap]: Is this part of changes for annotated tags?

Comment (christianguenter2) [reply src/zabapgit_git.prog.abap]: Yes, I did some experiments. It's not needed yet. Should I remove it?

**Comment by christianguenter2** [comment]
hm, looks like there are some merge commits. I'll try to cleanup...

File: src/zabapgit_git.prog.abap (hunk 1 above)
Comment (larshp) [reply src/zabapgit_git.prog.abap]: just keep it, I guess it prevents an error if the repo contains annotated tags

**Comment by larshp** [comment]
I can also do a "Squash and merge" if it is easier

**Comment by christianguenter2** [comment]
I did a rebase, now it looks better

**Comment by larshp** [comment]
merging to fix downport issue

**Pull request merged**